src/redecomp/RedecompMesh.cpp

```diff
@@ -77,27 +83,31 @@ RedecompMesh::RedecompMesh( const mfem::ParMesh& parent, double ghost_length,
       SLIC_ERROR_ROOT_IF( partitioner2d == nullptr, "Partitioner must be Partitioner2D." );
       auto partition_elems2d = dynamic_cast<const PartitionElements2D*>( partitioner2d->getPartitionEntity() );
       SLIC_ERROR_ROOT_IF( partition_elems2d == nullptr,
-                          "Redecomp requires the PartitionEntity "
-                          "to be PartitionElements." );
+                          "Redecomp requires the PartitionEntity to be PartitionElements." );
       break;
     }
     case 3: {
       auto partitioner3d = dynamic_cast<const Partitioner3D*>( partitioner.get() );
       SLIC_ERROR_ROOT_IF( partitioner3d == nullptr, "Partitioner must be Partitioner3D." );
       auto partition_elems3d = dynamic_cast<const PartitionElements3D*>( partitioner3d->getPartitionEntity() );
       SLIC_ERROR_ROOT_IF( partition_elems3d == nullptr,
-                          "Redecomp requires the PartitionEntity "
-                          "to be PartitionElements." );
+                          "Redecomp requires the PartitionEntity to be PartitionElements." );
       break;
     }
     default:
       SLIC_ERROR_ROOT( "Only 2D and 3D meshes are supported." );
   }
 
-  // preclude degenerate case where num elements < num ranks
-  auto n_parts = std::min( parent.GetNRanks(), static_cast<int>( parent.GetGlobalNE() ) );
+  auto parent_n_els = static_cast<int>( parent.GetGlobalNE() );
+  if ( n_ranks == 0 ) {
```

Comment:
It is hard for me to tell in github but this if-block and rank calculation appears twice. Why is that needed twice?

Reply:
This could probably be cleaned up, but there are currently two constructors that use this logic.